Remove interplay, prepare for release by tag

renatocaval opened the pull request:
Remove interplay, prepare for release by tag

Build on top of #335

ignasi35 commented on build.sbt:
```diff
@@ -60,49 +27,12 @@ Global / onLoad := (Global / onLoad).value.andThen { s =>
-
 lazy val twirl = project
   .in(file("."))
-  .enablePlugins(PlayRootProject)
```
I had to scratch my head on why `.enablePlugins(Common)` is not added here. But this looks good to me now.

Good job!

renatocaval replied:
I added Common, it's needed. I will push.

ignasi35 replied:
why is it needed?

There is no artifact to produce, there's no docs or sources in `"."`. What am I missing?

renatocaval replied:
You are right, I added too much. 

`validateCode` is failing because this projection doesn't have headers. It previously include it, but I removed. I don't need to add the Common plugin just for it. I will solve in another way.

renatocaval replied:
@ignasi35, I did another push. 

We now have a CopyrightHeader plugin automatically added everywhere. When we call `validateCode`, that's a command alias and it will require that all projects (also aggregation projects) to have the header configure. If not, the validate fails. 

I took the path of other projects in Akka in which we add such a plugin and we don't try to disable things in aggregation projects. 

We still have another issue, the validation may is failing, but travis is ignoring it. I don't want to fix it right now. It was already so. I prefer to run manually and move this PR forward. Later I fix it. I just want to avoid a side track that can become a rabbit hole.

ignasi35 commented on build.sbt:
```diff
@@ -60,49 +27,12 @@ Global / onLoad := (Global / onLoad).value.andThen { s =>
     crossScalaVersions := Nil, // workaround so + uses project-defined variants
-    mimaFailOnNoPrevious := false
+    mimaFailOnNoPrevious := false,
+    publish / skip := true
```
Wait, I think this needs an explicit `scalaVersion := Scala212` now. I think that's part of the workaround mentioned in the comment `// workaround so + uses project-defined variants` above.

ignasi35 replied:
I mean, it's there in the `Common` plugin ( in `override def globalSettings`) but that's not enabled here.

renatocaval replied:
I don't understand why this is needed. I will check if it will work to confirm it.

ignasi35 replied:
when using `+task` the default value is taken from the root project. Then , the fact that the root project has an empty `crossScalaVersions` also impacts how `+task` behaves.

It's a complete mess of cases and sub-cases. I learnt about all that while working on https://github.com/akka/akka-grpc/pull/907 (which is DRAFT for other reasonS.

renatocaval replied:
for the moment, I didn't observed any issue related to that. It just works, expect that `test` fails but because of something else. It's also happening in `master` so not related with this PR.

ignasi35 commented on build.sbt:
```diff
@@ -176,26 +102,27 @@ lazy val plugin = project
+// playBuildExtraTests := {
+//   (scripted in plugin).toTask("").value
+// }
+// playBuildExtraPublish := {
+//   (PgpKeys.publishSigned in plugin).value
+// }
```
This is not necessary since `pgp` signing will now happen on bintray. Not sure about the `scripted` part... looking.

renatocaval replied:
removed, still checking the scripted part

ignasi35 commented on project/Dependencies.scala:
```diff
@@ -0,0 +1,12 @@
+
+object Dependencies {
+
+  val Scala212      = "2.12.10"
```
```suggestion
  val Scala212      = "2.12.11"
```

ignasi35 commented on project/Dependencies.scala:
```diff
@@ -0,0 +1,12 @@
+
+  val PlayVersion = sys.props.getOrElse("play.version", sys.env.getOrElse("PLAY_VERSION", "2.8.0"))
+
+}
```
I'm guessing this means `master` was not releasing `sbt-0.13` versions of the `sbt-twirl` plugin, correct?

renatocaval replied:
yes

ignasi35 replied:
Just saw https://github.com/playframework/twirl/pull/335#discussion_r429953809

ignasi35 commented on build.sbt:
```diff
@@ -60,49 +27,12 @@ Global / onLoad := (Global / onLoad).value.andThen { s =>
-      pushChanges
-    )
-  }
-)
```
I think this was lost on the refactor.
Should this be on `project/Publish.scala`?

renatocaval replied:
We only publish to bintray (without promoting), we have nothing to push to git. So basically all this block can go.

ignasi35 replied:
Even the `+test`?

Ah, ofc! This will be a job in travis where tests have already run!

ignasi35 commented:
I like the split of `Header` out of `Common`. 👍🏼

ennru commented on build.sbt:
```diff
@@ -60,49 +27,12 @@ Global / onLoad := (Global / onLoad).value.andThen { s =>
-    releaseSettings,
     crossScalaVersions := Nil, // workaround so + uses project-defined variants
-    mimaFailOnNoPrevious := false
+    mimaFailOnNoPrevious := false,
```
Please do `.disablePlugins(MimaPlugin)` instead.

ennru commented on project/Dependencies.scala:
```diff
@@ -0,0 +1,12 @@
+object Dependencies {
+
+  val Scala212      = "2.12.11"
+  val Scala213      = "2.13.2"
```
.travis.yml is still on 2.13.1

ennru commented on project/Dependencies.scala:
```diff
@@ -0,0 +1,12 @@
+  val Scala213      = "2.13.2"
+  val ScalaVersions = Seq(Scala212, Scala213)
+
+  val PlayVersion = sys.props.getOrElse("play.version", sys.env.getOrElse("PLAY_VERSION", "2.8.0"))
```
Is this used?

ignasi35 commented:
@renatocaval I think there's [a setting missing](https://github.com/playframework/interplay/blob/master/src/main/scala/interplay/PlaySbtPluginBase.scala#L22) for sbt plugin projects affecting the `scripted` tests.

renatocaval commented:
@ennru / @ignasi35, it's now rebased with `master` and commits squashed. Also added the bintray credentials. 

I will run publish from my machine. This is ready for merge, only missing bit to review is the latest .travis.yml changes.

ignasi35 commented on build.sbt:
```diff
@@ -60,49 +27,12 @@ Global / onLoad := (Global / onLoad).value.andThen { s =>
-    mimaFailOnNoPrevious := false
+    publish / skip := true
   )
   .aggregate(apiJvm, apiJs, parser, compiler, plugin)
```
The `twirl` project doesn't declare (directly or via a plugin) a `scalaVersion`. It's probably fine since all projects get it declared via the `Common` plugin. It's only a bit surprising.

renatocaval replied:
I think it's fine since it's a aggregation project.

ignasi35 commented on build.sbt:
```diff
@@ -115,10 +45,9 @@ lazy val nodeJs = {
 lazy val api = crossProject(JVMPlatform, JSPlatform)
   .in(file("api"))
-  .enablePlugins(PlayLibrary, Playdoc)
+  .enablePlugins(Common, Playdoc, Omnidoc, PublishLibrary)
```
It's counterintuitive to have the `Playdoc` plugin on the `api` project and the `docs/` folder on the root folder.

It works because `Playdoc` uses a hardcoded path and piggybacks the `publish` task from `api` but `Playdoc` is completely unrelated to `api`. 

^^^ Not a comment, just a realization. 😒

renatocaval replied:
I'm fine to remove it if it doesn't make sense. I keep it because it was explicitly added to it, as you can see from the diff.

renatocaval commented:
I was able to publish from my machine to Bintray (snapshot repos)

![CleanShot 2020-05-26 at 16 13 16](https://user-images.githubusercontent.com/502982/82911677-7cf5f680-9f6c-11ea-9d8e-d9c08b4b6201.png)

![CleanShot 2020-05-26 at 16 12 43](https://user-images.githubusercontent.com/502982/82911690-80897d80-9f6c-11ea-9ff4-927a7333ba12.png)
